File: 프로그래머스/2/42577. 전화번호 목록/전화번호 목록.java
import java.util.*;

class Solution {
    public boolean solution(String[] phone_book) {
        
        Arrays.sort(phone_book);
        
        boolean answer = true;
        
        for(int i=1; i<phone_book.length; i++){
            String pre = phone_book[i-1];
            String now = phone_book[i];
            if(now.length() >= pre.length() && now.substring(0, pre.length()).equals(pre)){
                answer = false;
            }
        }
        
        return answer;
    }
}

File: 프로그래머스/2/42577. 전화번호 목록/전화번호 목록.js
function solution(phone_book) {
    
    phone_book.sort();
    
    var answer = true;
    
    for(let i=1; i<phone_book.length; i++){
        var pre = phone_book[i-1];
        var now = phone_book[i];
        if(pre.length <= now.length && now.substring(0, pre.length) == pre){
            answer = false;
        }
    }
   
    return answer;
}